include/veriblock/blockchain/blocktree.hpp

```diff
@@ -109,10 +109,32 @@ struct BlockTree {
     return acceptBlock(block, state, true);
   }
 
+  void invalidateBlockByHash(const hash_t& blockHash) {
+    index_t* blockIndex = getBlockIndex(blockHash);
+
+    for (auto chain_it = fork_chains_.begin();
+         chain_it != fork_chains_.end();) {
+      invalidateBlockFromChain(*chain_it, &blockIndex);
```

Comment:
In case common block of two branches is deleted one of the branches becomes orphaned.